File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.cpp
class Solution {
    private:
    bool isVowel(char c)
    {
        char a=tolower(c);
        if(a=='a' || a=='e' || a=='i' || a=='o' || a=='u')
        {
            return true;
        }
        return false;
    }
public:
    string reverseVowels(string s) {

        int l=0;
        int r=s.size()-1;
        while(l<r)
        {
            

           while(l<r && !isVowel(s[r])) r--;
           while(l<r && !isVowel(s[l])) l++;
           if(l<r)
            {
                swap(s[l],s[r]);
                l++;
                r--;
            }

        //    cout<<s<<endl;
        }
        return s;
    }
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:


        l=0
        r=len(s)-1
        s=list(s)


        while l<r:



            if s[l].lower() in 'aeiou' and s[r].lower() in 'aeiou':
                s[l],s[r]=s[r],s[l]
                l+=1
                r-=1

            while l<r and s[l].lower() not in 'aeiou':
                l+=1
            while l<r and  s[r].lower() not in 'aeiou':
                r-=1


        return ''.join(s)

        